src/Form/PasswordChange.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Password\PasswordInterface;
+use Drupal\Core\Session\SessionManagerInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Password change form.
+ */
+class PasswordChange extends FormBase {
+
+  /**
+   * Date time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $dateTime;
+
+  /**
+   * Current user.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $user;
+
+  /**
+   * Password service.
+   *
+   * @var \Drupal\Core\Password\PasswordInterface
+   */
+  protected $password;
+
+  /**
+   * Session manager.
+   *
+   * @var \Drupal\Core\Session\SessionManagerInterface
+   */
+  protected $sessionManager;
+
+  /**
+   * Constructs the password change form.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $date_time
+   *   Date time.
+   * @param \Drupal\user\UserInterface $user
+   *   Current user.
+   * @param \Drupal\Core\Password\PasswordInterface $password
+   *   Password service.
+   * @param \Drupal\Core\Session\SessionManagerInterface $session_manager
+   *   Session manager.
+   */
+  public function __construct(TimeInterface $date_time, UserInterface $user, PasswordInterface $password, SessionManagerInterface $session_manager) {
+    $this->dateTime = $date_time;
+    $this->user = $user;
+    $this->password = $password;
+    $this->sessionManager = $session_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): PasswordChange {
+    $user_id = $container->get('current_user')->id();
+    return new static(
+      $container->get('datetime.time'),
+      $container->get('entity_type.manager')->getStorage('user')->load($user_id),
```

Comment:
Quoted from Slack last week:

> It just came into my mind after I submitted my latest review comments. Do not loaf the full user object in a service construct. If the cache is cold it could slow down the build of the service container and who knows what else it could cause. Load the user object of the current user where it is actually needed,in a method, which only gets called when it is necessary.
If you really insist the current solution then maybe transforming these services to lazy ones could solve this bottleneck, but that would be just a workaround.